tasks/pkg-suse.yml

```diff
@@ -0,0 +1,59 @@
+---
+- block: # Work around due to SNI check for SLES11
+  - name: Stat if RPM key already exist
+    stat:
+      path: /tmp/DATADOG_RPM_KEY.public
+    register: ddkey
+  - name: Download new RPM key
+    shell: "curl {{ datadog_zypper_gpgkey }} -o /tmp/DATADOG_RPM_KEY.public"
+    args:
+      warn: no
+    when: not ddkey.stat.exists
+  when: ansible_distribution_version|int == 11
+
+- name: Download new RPM key
+  get_url:
+    url: "{{ datadog_zypper_gpgkey }}"
+    dest: /tmp/DATADOG_RPM_KEY.public
+    sha256sum: "{{ datadog_zypper_gpgkey_sha256sum }}"
+  when: ansible_distribution_version|int >= 12
+
+- name: Import new RPM key
+  rpm_key:
+    key: /tmp/DATADOG_RPM_KEY.public
+    state: present
+  when: not ansible_check_mode
+
+- name: Install DataDog zypper repo
+  zypper_repository:
+    name: datadog
+    description: Datadog, Inc.
+    repo: "{{ datadog_zypper_repo }}"
+    enabled: yes
+    auto_import_keys: yes
+    disable_gpg_check: "{% if ansible_distribution_version|int == 11 %}yes{% else %}no{% endif %}" # SLES11 considers datadog's repo as untrusted
+    state: "{% if datadog_agent5 %}absent{% else %}present{% endif %}"
+
+- name: (agent5) Install DataDog yum repo
+  zypper_repository:
+    name: datadog
+    description: Datadog, Inc.
+    repo: "{{ datadog_agent5_zypper_repo }}"
+    enabled: yes
+    auto_import_keys: yes
+    disable_gpg_check: "{% if ansible_distribution_version|int == 11 %}yes{% else %}no{% endif %}" # SLES11 considers datadog's repo as untrusted
+    state: "{% if datadog_agent5 %}present{% else %}absent{% endif %}"
+
+- name: Install pinned datadog-agent package
+  zypper:
+    name: "datadog-agent={{ datadog_agent_version }}"
+    state: present
```

Comment:
When downgrading from agent6 to agent5 zypper will refuse to install the agent since the version is lower.

We need to set the `oldpackage` option:
```yaml
oldpackage: "{{ datadog_agent_allow_downgrade }}"
```

Reply:
This is fixed using the `force` param in the zypper module.

Reply:
After further testing, it looks like SLES11's Zypper ignores the `force` param. In order to downgrade an agent from 6 to 5, the agent needs to be uninstalled first before installing 5. I can either make this node in the README.md file or create a tasks to uninstall current agent first with `datadog_agent_allow_downgrade` conditions then install v5. Let me know what works for you guys.

Thanks